source/modulo_core/include/modulo_core/JointPositionsBroadcaster.hpp

```diff
@@ -0,0 +1,97 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <modulo_interfaces/msg/joint_positions.hpp>
+#include <modulo_interfaces/msg/joint_positions_collection.hpp>
+#include <state_representation/space/joint/JointPositions.hpp>
+
+namespace modulo_core {
+
+class JointPositionsBroadcaster {
+public:
+  template<class NodeT, class AllocatorT = std::allocator<void>>
+  JointPositionsBroadcaster(
+      NodeT&& node, const rclcpp::QoS& qos = rclcpp::QoS(1).transient_local(),
+      const rclcpp::PublisherOptionsWithAllocator<AllocatorT>& options =
+          []() {
+            rclcpp::PublisherOptionsWithAllocator<AllocatorT> options;
+            options.qos_overriding_options = rclcpp::QosOverridingOptions{
+                rclcpp::QosPolicyKind::Depth, rclcpp::QosPolicyKind::History, rclcpp::QosPolicyKind::Reliability};
+            /*
+              This flag disables intra-process communication when the JointPositionsBroadcaster is constructed
+              using an existing node handle which happens to be a component (in rclcpp terminology).
+              Required until rclcpp intra-process communication supports transient_local QoS durability.
+            */
+            options.use_intra_process_comm = rclcpp::IntraProcessSetting::Disable;
+            return options;
+          }())
+      : JointPositionsBroadcaster(
+            rclcpp::node_interfaces::get_node_parameters_interface(node),
+            rclcpp::node_interfaces::get_node_topics_interface(node), qos, options) {}
+
+  template<class AllocatorT = std::allocator<void>>
+  JointPositionsBroadcaster(
+      rclcpp::node_interfaces::NodeParametersInterface::SharedPtr node_parameters,
+      rclcpp::node_interfaces::NodeTopicsInterface::SharedPtr node_topics,
+      const rclcpp::QoS& qos = rclcpp::QoS(1).transient_local(),
+      const rclcpp::PublisherOptionsWithAllocator<AllocatorT>& options = []() {
+        rclcpp::PublisherOptionsWithAllocator<AllocatorT> options;
+        options.qos_overriding_options = rclcpp::QosOverridingOptions{
+            rclcpp::QosPolicyKind::Depth, rclcpp::QosPolicyKind::History, rclcpp::QosPolicyKind::Reliability};
+        /*
+          This flag disables intra-process communication when the JointPositionsBroadcaster is constructed
+          using an existing node handle which happens to be a component (in rclcpp terminology).
```

Comment:
nit: here I guess this not entirely accurate since we only have the parameters interface?